code/modules/wiremod/datatypes/composite/composite.dm

```diff
@@ -85,12 +88,19 @@
 	src.datatype = datatype
 	src.base_datatype = base_datatype
 	src.composite_datatypes = composite_datatypes
+	src.primary_composited_datatype = generate_primary_composite()
 	abstract = FALSE
 
 	composite_datatypes_style += list()
 	for(var/datatype_to_check in composite_datatypes)
 		composite_datatypes_style += GLOB.circuit_datatypes[datatype_to_check].color
 
+/datum/circuit_datatype/composite_instance/proc/generate_primary_composite()
+	if (base_datatype == PORT_COMPOSITE_TYPE_LIST)
+		return composite_datatypes[1]
+	else
+		return composite_datatypes[2] // assoc, the 2nd is our non-key
```

Comment:
same thing here. I'm against the typecasting part of this in its entirety since it defeats the purposes of having types like these. Typecast is better kept to primitives instead of trying to hardcode what composites are allowed to be.
Composites are not necessarily just built to be regular lists and assoc lists, they can be any sort of type that they want, composites are just a framework to let regular lists and assoc lists easily exist.

This sort of hardcoding of what a composite is allowed to be is completely against why the typing system exists in the first place.